vnext/Microsoft.ReactNative/Views/TextInputViewManager.cpp

```diff
@@ -373,6 +362,61 @@ void TextInputShadowNode::registerEvents() {
       true);
 }
 
+void TextInputShadowNode::registerPreviewKeyDown() {
+  auto control = GetView().as<xaml::Controls::Control>();
+  auto tag = m_tag;
+  m_controlKeyDownRevoker =
+      control.PreviewKeyDown(winrt::auto_revoke, [=](auto &&, xaml::Input::KeyRoutedEventArgs const &args) {
+        auto isMultiline = m_isTextBox && control.as<xaml::Controls::TextBox>().AcceptsReturn();
+        auto shouldSubmit = !args.Handled();
```

Comment:
When would `shouldSubmit` be false at this point?

Reply:
That would be if the event was marked as handled inside `HandledKeyboardEventHandler::KeyboardEventHandledHandler`, right?

Reply:
If so, does using `submitKeyEvents` necessitate specifying the same keys in `keyDownEvents` for this to work?

Reply:
shouldSubmit will be false when the routed event is marked handled for *any* reason (`keyDownEvents` setting, mixed-mode child control setting the event to Handled, etc.)

It does not necessitate specifying the same keys in `keyDownEvents`. In fact, declaring the key in `keyDownEvents` for `handledEventPhase: Capturing` will prevent the `submitKeyEvents` key from firing (because the event will be marked handled). This is expected behavior. We actually have a scenario that requires this in our app. The multiline TextInput sets `submitKeyEvents={[{code: 'Enter'}]}` and then in certain scenarios (e.g., an @ reference flyout), we set `keyDownEvents={[{code: 'Enter', handledEventPhase: Capturing}]}` so we can temporarily disable submit and route the 'Enter' event key press event to select an @ reference target.  

I agree that it's worthwhile documenting that `keyDownEvents` takes precedence over `submitKeyEvents`. In fact, we

Reply:
Thanks for the interesting example! 

As for the documentation, the spec file in the repo would be one place. We will still want to put this in the regular docs. (We aspire to do this at PR time, but realistically, anytime before 0.65 is released; you can create an issue to track that)